lib/processor.js

```diff
@@ -58,14 +61,23 @@ function preprocess(text) {
 }
 
 /**
+ * Excludes unsatisfiable rules from the list of messages.
+ * @param {Message} message A message from the linter.
+ * @returns {boolean} True if the message should be included in output.
+ */
+function excludeUnsatisfiableRules(message) {
+    return UNSATISFIABLE_RULES.indexOf(message.ruleId) < 0;
```

Comment:
If you are okay with dropping support for Node < 4 , we can use a `Set` instead of an array and have quicker (and simpler) lookups.